File: src/main/scala-graphql/io/github/graphql/model/RegistryPackageStatisticsResponseProjection.scala
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection

/**
 * Response projection for RegistryPackageStatistics
 */
@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:01+0800"
)
class RegistryPackageStatisticsResponseProjection extends GraphQLResponseProjection {

    override def all$(): RegistryPackageStatisticsResponseProjection = all$(3)

    override def all$(maxDepth: Int): RegistryPackageStatisticsResponseProjection = {
        this.downloadsThisMonth()
        this.downloadsThisWeek()
        this.downloadsThisYear()
        this.downloadsToday()
        this.downloadsTotalCount()
        this.typename()
        this
    }

    def downloadsThisMonth(): RegistryPackageStatisticsResponseProjection = {
        downloadsThisMonth(null.asInstanceOf[String])
    }

    def downloadsThisMonth(alias: String): RegistryPackageStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("downloadsThisMonth").alias(alias))
        this
    }

    def downloadsThisWeek(): RegistryPackageStatisticsResponseProjection = {
        downloadsThisWeek(null.asInstanceOf[String])
    }

    def downloadsThisWeek(alias: String): RegistryPackageStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("downloadsThisWeek").alias(alias))
        this
    }

    def downloadsThisYear(): RegistryPackageStatisticsResponseProjection = {
        downloadsThisYear(null.asInstanceOf[String])
    }

    def downloadsThisYear(alias: String): RegistryPackageStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("downloadsThisYear").alias(alias))
        this
    }

    def downloadsToday(): RegistryPackageStatisticsResponseProjection = {
        downloadsToday(null.asInstanceOf[String])
    }

    def downloadsToday(alias: String): RegistryPackageStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("downloadsToday").alias(alias))
        this
    }

    def downloadsTotalCount(): RegistryPackageStatisticsResponseProjection = {
        downloadsTotalCount(null.asInstanceOf[String])
    }

    def downloadsTotalCount(alias: String): RegistryPackageStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("downloadsTotalCount").alias(alias))
        this
    }

    def typename(): RegistryPackageStatisticsResponseProjection = {
        typename(null.asInstanceOf[String])
    }

    def typename(alias: String): RegistryPackageStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("__typename").alias(alias))
        this
    }


}

File: src/main/java-graphql/io/github/graphql/model/RegistryPackageStatisticsResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for RegistryPackageStatistics
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class RegistryPackageStatisticsResponseProjection extends GraphQLResponseProjection {

    public RegistryPackageStatisticsResponseProjection() {
    }

    @Override
    public RegistryPackageStatisticsResponseProjection all$() {
        return all$(3);
    }

    @Override
    public RegistryPackageStatisticsResponseProjection all$(int maxDepth) {
        this.downloadsThisMonth();
        this.downloadsThisWeek();
        this.downloadsThisYear();
        this.downloadsToday();
        this.downloadsTotalCount();
        this.typename();
        return this;
    }

    public RegistryPackageStatisticsResponseProjection downloadsThisMonth() {
        return downloadsThisMonth(null);
    }

    public RegistryPackageStatisticsResponseProjection downloadsThisMonth(String alias) {
        fields.add(new GraphQLResponseField("downloadsThisMonth").alias(alias));
        return this;
    }

    public RegistryPackageStatisticsResponseProjection downloadsThisWeek() {
        return downloadsThisWeek(null);
    }

    public RegistryPackageStatisticsResponseProjection downloadsThisWeek(String alias) {
        fields.add(new GraphQLResponseField("downloadsThisWeek").alias(alias));
        return this;
    }

    public RegistryPackageStatisticsResponseProjection downloadsThisYear() {
        return downloadsThisYear(null);
    }

    public RegistryPackageStatisticsResponseProjection downloadsThisYear(String alias) {
        fields.add(new GraphQLResponseField("downloadsThisYear").alias(alias));
        return this;
    }

    public RegistryPackageStatisticsResponseProjection downloadsToday() {
        return downloadsToday(null);
    }

    public RegistryPackageStatisticsResponseProjection downloadsToday(String alias) {
        fields.add(new GraphQLResponseField("downloadsToday").alias(alias));
        return this;
    }

    public RegistryPackageStatisticsResponseProjection downloadsTotalCount() {
        return downloadsTotalCount(null);
    }

    public RegistryPackageStatisticsResponseProjection downloadsTotalCount(String alias) {
        fields.add(new GraphQLResponseField("downloadsTotalCount").alias(alias));
        return this;
    }

    public RegistryPackageStatisticsResponseProjection typename() {
        return typename(null);
    }

    public RegistryPackageStatisticsResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}
